services/city.service.go

```diff
@@ -0,0 +1,28 @@
+package services
+
+import (
+	"fmt"
+	"project_weather/generated/dao/model"
+)
+
+type CityService struct{}
+
+func NewCityService() *CityService {
+	return &CityService{}
+}
+
+func (s *CityService) FormatWeatherDataForCity(weatherData map[string]interface{}) *model.City {
+	name := weatherData["city"].(map[string]interface{})["name"].(string)
```

Comment:
Each dot in the following 4 lines is possible is potential runtime error
e.g. 1st line:
- you have to check if `weatherData["city"]` exists like `if wd, exists := weatherData["city"]; !exists { ... }`
- you have to check whether the typecasting is ok like `if wdMap, ok := wd.(map[string]interface{}); !ok { ... }`
... and again if name exists in the new map and then if its string...
This code is a ticking time bomb :smile: ... I would rely on some generated models instead of nested `map[string]interface{}`